modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.es.js

```diff
@@ -70,9 +71,11 @@ export default function render(renderable, renderData, container) {
 
 		// eslint-disable-next-line @liferay/portal/no-react-dom-render
 		ReactDOM.render(
-			<ClayIconSpriteContext.Provider value={spritemap}>
-				{Component ? <Component {...renderData} /> : renderable}
-			</ClayIconSpriteContext.Provider>,
+			<ClayTooltipProvider>
```

Comment:
This looks good to me. I was just wondering if @matuzalemsteles or @wincent can think of a reason we decided not to do this yet. Maybe it just didn't come up.

Reply:
Yep, as far as I know, we never even discussed this possibility, but it seems fine to do it.

Reply:
Well, I think the initial idea of this was to be the Provider of the application but probably people were always adding above the component with Title for some reason... I remember we had an [issue open](https://github.com/liferay/clay/issues/2353) in Clay where we would centralize all Providers, something like ClayManager to probably add to the render. This would contain:

- ClayModalProvider
- ClayIconSpriteContext
- ClayTooltipProvider

I think we didn't get to prioritize that because we haven’t seen a great need at the moment. This could alleviate forgetting to add any Provider.

Perhaps another reason was that Tooltip was also always controlled by singleton on the Portal and we had been working on his evolution to use React, maybe that discouraged us from using Provider since it only works on the React tree.